sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/DefaultTaskKiller.java

```diff
@@ -22,6 +22,10 @@ public DefaultTaskKiller(TaskFailureListener taskFailureListener, SchedulerDrive
 
     @Override
     public void killTask(TaskID taskId, boolean destructive) {
+        if (taskId.getValue().isEmpty()) {
+            return;
```

Comment:
If this isn't expected, maybe log an error (or just throw). If it is expected, maybe describe the case where it may happen.